crates/guest-rust/rt/src/async_support/stream_support.rs

```diff
@@ -480,20 +480,20 @@ where
         (reader, mut buf, cleanup): Self::InProgress,
         code: u32,
     ) -> Result<Self::Result, Self::InProgress> {
-        match code {
-            results::BLOCKED => Err((reader, buf, cleanup)),
+        match ReturnCode::decode(code) {
+            ReturnCode::Blocked => Err((reader, buf, cleanup)),
 
             // Note that the `cleanup`, if any, is discarded here.
-            results::CLOSED => Ok((StreamResult::Closed, buf)),
+            ReturnCode::Closed(0) => Ok((StreamResult::Closed, buf)),
 
             // When an in-progress read is successfully cancelled then the
             // allocation that was being read into, if any, is just discarded.
             //
             // TODO: should maybe thread this around like `AbiBuffer` to cache
             // the read allocation?
-            results::CANCELED => Ok((StreamResult::Cancelled, buf)),
+            ReturnCode::Cancelled(0) => Ok((StreamResult::Cancelled, buf)),
 
-            amt => {
+            ReturnCode::Completed(amt) | ReturnCode::Closed(amt) | ReturnCode::Cancelled(amt) => {
```

Comment:
I'll note that for now I've opted to just lose information for here as it's, in theory harmless. For example if a stream is closed after transferring N items then future writes should in theory not even hit the host and immediately say "nope it's close", or perhaps the state is reflected in the return value where the stream is consumed. I don't think it's super important to expose these states just yet so I've left that for a future update.